otter/tap/api.py

```diff
@@ -278,17 +279,22 @@ def on_client_ready(_):
                                              get_service_configs(config),
                                              kz_client, store, supervisor,
                                              cassandra_cluster)
+
             # Setup scheduler service after starting
             scheduler = setup_scheduler(parent, dispatcher, store, kz_client)
-            health_checker.checks['scheduler'] = scheduler.health_check
-            otter.scheduler = scheduler
+            if scheduler is not None:
```

Comment:
This is making me deeply suspicious - I don't fully understand what's going on, but all I see in this PR right now is
```
+        mock_setup_scheduler.return_value = None
```
I don't see anywhere that `setup_scheduler` itself has been modified to return a different value.

If it really needs to return a new type, could you consider an Active Nothing here?